tests/phpunit/CRM/Contribute/Form/Contribution/MainTest.php

```diff
@@ -80,40 +80,44 @@ public function testSetRecurFunctionNotAvailable() {
    *
    * @return \CRM_Contribute_Form_Contribution_Main
    */
-  protected function getContributionForm() {
-    $form = new CRM_Contribute_Form_Contribution_Main();
-    $form->_values['is_monetary'] = 1;
-    $form->_values['is_pay_later'] = 0;
-    $form->_priceSetId = $this->callAPISuccessGetValue('PriceSet', [
+  protected function getContributionForm($params = []) {
+    $params['priceSetID'] = $params['priceSetID'] ?? $this->callAPISuccessGetValue('PriceSet', [
       'name' => 'default_membership_type_amount',
       'return' => 'id',
     ]);
-    $priceFields = $this->callAPISuccess('PriceField', 'get', ['id' => $form->_priceSetId]);
-    $form->_priceSet['fields'] = $priceFields['values'];
-    $paymentProcessorID = $this->paymentProcessorCreate(['payment_processor_type_id' => 'Dummy']);
-    $form->_paymentProcessor = [
-      'billing_mode' => CRM_Core_Payment::BILLING_MODE_FORM,
-      'object' => Civi\Payment\System::singleton()->getById($paymentProcessorID),
-      'is_recur' => TRUE,
-    ];
-    $form->_values = [
-      'title' => "Test Contribution Page",
-      'financial_type_id' => 1,
+
+    $contributionPageParams = (array_merge($params, [
       'currency' => 'NZD',
       'goal_amount' => 6000,
-      'is_pay_later' => 1,
-      'is_monetary' => TRUE,
+      'is_pay_later' => 0,
+      'is_monetary' => 1,
       'pay_later_text' => 'Front up',
       'pay_later_receipt' => 'Ta',
-    ];
+      'is_email_receipt' => 1,
+      'payment_processor' => $this->paymentProcessorCreate([
+        'payment_processor_type_id' => 'Dummy',
+        'is_test' => 0,
+      ]),
+      'amount_block_is_active' => 1,
+    ]));
+
+    /** @var \CRM_Contribute_Form_Contribution_Main $form */
+    $form = $this->getFormObject('CRM_Contribute_Form_Contribution_Main');
+    $contributionPage = reset($this->contributionPageCreate($contributionPageParams)['values']);
+    $form->set('id', $contributionPage['id']);
+    // PriceSetId gets looked up by CRM_Price_BAO_PriceSet::getFor() in ContributionBase::preProcess so we need to
+    // set it here:
+    CRM_Core_DAO::executeQuery("INSERT INTO civicrm_price_set_entity (entity_table, entity_id, price_set_id)
+            VALUES ('civicrm_contribution_page', {$contributionPage['id']}, {$params['priceSetID']})");
```

Comment:
I'm not sure how/when this is set on a real contribution page. There is no API for price_set_entity but this is required for the contributionPage to load the priceSet.

Reply:
@mattwire maybe https://github.com/civicrm/civicrm-core/blob/master/CRM/Contribute/Form/ContributionPage/Amount.php#L568

Reply:
Thanks, that looks like the one!